lib/susedistribution.pm

```diff
@@ -203,10 +203,8 @@ sub init_consoles {
             $s390_params =~ s,\@S390_HOST\@,$s390_host,g;
             set_var("S390_NETWORK_PARAMS", $s390_params);
 
-            my @s390_params = split / /, $s390_params;
-            my %s390_params = map { split "=" } @s390_params;
-
-            $hostname = $s390_params{Hostname};
+            $hostname = $s390_params;
+            $hostname =~ s/.*Hostname=(\S*)\s.*/$1/;
```

Comment:
This will fail if hostname is the last config parameter, e.g. "S390_FOO=bar Hostname=z42".
```
$hostname =~ s/.*Hostname=(\S*).*/$1/;
```
should work just fine. No need to check for whitespace after the hostname when the hostname itself is matching all non-whitespace.